crates/runtime/src/memory.rs

```diff
@@ -212,33 +212,14 @@ pub enum Memory {
     Dynamic(Box<dyn RuntimeLinearMemory>),
 }
 
-fn memory_growing(
-    limiter: &mut Option<&mut dyn ResourceLimiter>,
-    current: usize,
-    desired: usize,
-    maximum: Option<usize>,
-) -> bool {
-    match limiter {
-        Some(ref mut l) => l.memory_growing(current, desired, maximum),
-        None => true,
-    }
-}
-
-fn memory_grow_failed(limiter: &mut Option<&mut dyn ResourceLimiter>, error: &Error) {
-    match limiter {
-        Some(l) => l.memory_grow_failed(error),
-        None => {}
-    }
-}
-
 impl Memory {
     /// Create a new dynamic (movable) memory instance for the specified plan.
     pub fn new_dynamic(
         plan: &MemoryPlan,
         creator: &dyn RuntimeMemoryCreator,
-        limiter: Option<&mut dyn ResourceLimiter>,
+        store: &mut Option<&mut dyn Store>,
```

Comment:
With taking a store here instead of a limiter I think it's actually a bug in Wasmtime if we ever create a table/memory *without* a store because that means if there were a registered limiter we'd be skipping that. I think we probably want to change this constructor to take `&mut dyn Store` instead of `Option<_>` to reflect that and we can fix it if it ever trips a panic (I don't believe it will checking Wasmtime's current usage).

Reply:
I think changing this to not-an-option will also propagate out relatively far-ish.  That's ok though because almost all instance allocation requests should have a store configured, only the initial blank module as well as host functions should skip their stores during construction, and the `Option`-handling is basically just here to make sure those two cases don't panic/segfault.

Reply:
Yep my initial constructor took just a `&mut dyn Store` and the option fell out of some complications elsewhere, and I hadn't thought that all the way through to indicate a bug in Wasmtime. I'll have to chase that out.